Merge_Sort_C/main.c

```diff
@@ -0,0 +1,78 @@
+#include <stdio.h>
+
+void PrintArray(int array[], int sizeOfArray)
+{
+    for(int i = 0; i < sizeOfArray; i++)
+    {
+        printf("%d  ", array[i]);
+    }
+    printf("\n");
+}
+
+void MergeArrays(int array[], int leftHalf[], int rightHalf[], int left, int right)
+{
+    int i = 0;
+    int j = 0;
+    int k = 0;
+
+    while (i < left && j < right) {
+        if(leftHalf[i] <= rightHalf[j])
+        {
+            array[k++] = leftHalf[i++];
+        }
+        else
+        {
+            array[k++] = rightHalf[j++];
+        }
+    }
+    while (i < left)
+    {
+        array[k++] = leftHalf[i++];
+    }
+    while (j < right)
+    {
+        array[k++] = rightHalf[j++];
+    }
+}
+
+void SortArray(int array[], int sizeOfArray)
+{
+    if(sizeOfArray < 2) return;
+
+    int beforeMiddle = sizeOfArray / 2;
+    int afterMiddle = sizeOfArray - beforeMiddle;
+    int leftHalf[beforeMiddle];
+    int rightHalf[afterMiddle];
+
+    for(int i = 0; i < beforeMiddle; i++)
```

Comment:
This is a perfect example for a `[Method extraction](https://refactoring.guru/uk/extract-method)` refactoring.

Reply:
Suggested method signature: void CopyArray(int source[], int destination[], int startIndex, int count)
`int source[]` - array to copy from
`int destination[]` - array to copy to
`startIndex` - index in the source array from which to start copying
`int count` - the number of elements to copy